### ISSUE TITLE
[High Priority] Latest Installer launches simplewall without first asking user.

### ISSUE THREAD

**Issue by gettysburg** [henrypp/simplewall#1119]
Hi,

I am currently updating simplewall to version 3.6.1 on my systems, and noticed the installer now always launches the extracted binary, right after unpacking all files - this happens before the user has a chance to click continue, i.e. before the "Completing simplewall Setup" page appears:

![image](https://user-images.githubusercontent.com/2256003/145684432-39dd4e76-a095-4c25-844f-fd103797667e.png)

The checkbox named "Run simplewall" is checked by default, so if the user now exits setup by clicking finish, the installer effectively tries to spawn another instance of simplewall, which obviously won't work.

**Comment by Pandaa-P** [comment]
Happened with me as well.  simplewall runs before the installer finished the Setup.

**Comment by Roddly** [comment]
I consider this behavior as a major issue with the **silent installation**. Unless you add a parameter in the command line, simplewall should never start after a silent installation.

Also, the silent installation never exits:

```
$list = @{
    FilePath = 'simplewall-3.6.1-setup.exe'
    ArgumentList = '/S'
    Verb = 'RUNAS'
    Wait = $true
    Passthru = $true
}
$p = Start-Process @list
# never reached (unless you close simplewall.exe)
$result.Add('exit_code', $p.ExitCode)
```

**Comment by gettysburg** [comment]
@Roddly  I did not know about that that but now that I do, I suspect probably some badly damaged installer code or an update of the installer framework gone wrong, which for some reason waits for the child process to exit, I changed the title and will ping the developer so he gets a notification.

@henrypp

Please, look into this, and then, if you still have some time, check out the other issue I opened, too:

#1126

**Comment by Roddly** [comment]
The problem still remains in version 3.6.2.

**Comment by gettysburg** [comment]
I can confirm that this issue still exists after just upgrading, it even appears in the NSIS log:

![image](https://user-images.githubusercontent.com/2256003/165346848-381cea75-c20d-4a59-b61c-8b88fd54ec30.JPG)

**Comment by Pandaa-P** [comment]
It's most likely a bug from `NSIS` maybe use `InnoSetup` instead?

**Comment by gettysburg** [comment]
It's not an NSIS bug, we use NSIS for all our software and it doesn't behave this way.

**Comment by gettysburg** [comment]
@Pandaa-P Not possible, NSIS hasn't been updated since September 2021 so there is no way for this bug to have been introduced.

**Comment by Pandaa-P** [comment]
It could be an old bug and we suffer from it, it's not the first time I encounter a software that run before the installer end the setup process.

**Comment by gettysburg** [comment]
Henry would have needed to change the NSIS configuration in order to trigger an old bug, so I don't think it's that either.

Anyway, let him take a look at it and he should be able to fix it.

@henrypp

**Comment by Pandaa-P** [comment]
Was this bug exist in old versions? (I'm not an old user of simplewall, I knew about simplewall late last year)
If the old versions was clean from this bug, we can trace which version of simplewall the bug was introduced, and what change caused it.

**Comment by Roddly** [comment]
> Was this bug exist in old versions?

Last working version is 3.5.3. The problem exists since version 3.6.

**Comment by gettysburg** [comment]
Fixed now in v3.6.3 installer.
